controllers/k8ssandra/k8ssandracluster_controller_test.go

```diff
@@ -768,6 +768,7 @@ func parseCassandraConfig(config *api.CassandraConfig, serverVersion string, sys
 	)
 	template := cassandra.DatacenterConfig{
 		ServerVersion:   semver.MustParse(serverVersion),
+		ServerType:      "cassandra",
```

Comment:
We shouldn't require this in k8ssandra-operator, breaks backwards compatibility. This test should pass without this value ("cassandra" should be the assumed default).

Reply:
It does not break backwards compatibility. 

The API field is `CassandraClusterTemplate.ServerType` and the field is annotated with `+kubebuilder:default=cassandra`. So it cannot be empty in real life.

The field `dcConfig.ServerType` is an internal field. It is populated in the `Coalesce` function, see [here](https://github.com/k8ssandra/k8ssandra-operator/blob/16020d6ab308e9b6b2f3a947078e24415d21663b/pkg/cassandra/datacenter.go#L284). So IRL it cannot be empty either, which is btw enforced [here](https://github.com/k8ssandra/k8ssandra-operator/blob/16020d6ab308e9b6b2f3a947078e24415d21663b/pkg/cassandra/datacenter.go#L135-L137).

Here we are hacking the internal API to produce a JSON that will be used to compare with the actual JSON stored in `CassandraDatacenter.Spec.Config`. That's why we need to set `ServerType` manually.